src/main/kotlin/graphics/scenery/Arrow.kt

```diff
@@ -0,0 +1,168 @@
+package graphics.scenery
+
+import cleargl.GLVector
+import graphics.scenery.backends.ShaderType
+import java.nio.FloatBuffer
+import java.nio.IntBuffer
+
+/**
+ * Class for creating 3D arrows, derived from [Node] and using [HasGeometry].
+ * The arrow is defined with [vector] having the vector's foot/start as the
+ * reference coordinate (arrow's foot will appear at the coordinate given to setPosition()).
+ *
+ *   /|\
+ *  / | \
+ * /--+--\
+ *    |
+ *    |
+ *    |
+ * The arrow is created as the main segment (drawn vertically bottom to up),
+ * then 3 segments to build a triangle and then another triangle perpendicular
+ * to the former one; altogehter 7 segments drawn sequentially.
+ * The arrow head scales with the length of the arrow.
+ *
+ * @author Vladimir Ulman <ulman@mpi-cbg.de>
+ */
+class Arrow(var vector: GLVector) : Node("Arrow"), HasGeometry {
+    /** Size of one vertex (e.g. 3 in 3D) */
+    override val vertexSize: Int = 3
+    /** Size of one texcoord (e.g. 2 in 3D) */
+    override val texcoordSize: Int = 2
+    /** Geometry type -- Default for Line is [GeometryType.LINE] */
+    override var geometryType: GeometryType = GeometryType.LINE_STRIP_ADJACENCY
+    /** Vertex buffer */
+    override var vertices: FloatBuffer = BufferUtils.allocateFloat(30)
```

Comment:
These guys might need resizing, have a look at `Line` for some code.

Reply:
Well... the methods that can modify them, that is `addPoint()` and `clearPoints()` are made private to make sure the only way to change the content of the `vertices` & friends attributes happens via the method `reshape()`, which however sets always exactly 10 points (10*3 = 30).  

I see an issue, however, if the attributes `vertices` & friends are public, not protected...